contracts/cw3-flex-multisig/src/contract.rs

```diff
@@ -0,0 +1,993 @@
+use std::cmp::Ordering;
+
+use cosmwasm_std::{
+    attr, to_binary, Binary, BlockInfo, CanonicalAddr, CosmosMsg, Deps, DepsMut, Empty, Env,
+    HandleResponse, HumanAddr, InitResponse, MessageInfo, Order, StdResult,
+};
+
+use cw0::{maybe_canonical, Expiration};
+use cw2::set_contract_version;
+use cw3::{
+    ProposalListResponse, ProposalResponse, Status, ThresholdResponse, Vote, VoteInfo,
+    VoteListResponse, VoteResponse, VoterListResponse, VoterResponse,
+};
+use cw4::Cw4Contract;
+use cw_storage_plus::Bound;
+
+use crate::error::ContractError;
+use crate::msg::{HandleMsg, InitMsg, QueryMsg};
+use crate::state::{next_id, parse_id, Ballot, Config, Proposal, BALLOTS, CONFIG, PROPOSALS};
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:cw3-flex-multisig";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+pub fn init(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InitMsg,
+) -> Result<InitResponse, ContractError> {
+    if msg.required_weight == 0 {
+        return Err(ContractError::ZeroWeight {});
+    }
+    // we just convert to canonical to check if this is a valid format
+    if deps.api.canonical_address(&msg.group).is_err() {
+        return Err(ContractError::InvalidGroup {});
+    }
+
+    let group = Cw4Contract(msg.group);
+    let total_weight = group.total_weight(&deps.querier)?;
+
+    if total_weight < msg.required_weight {
```

Comment:
Given that `total_weight` is now dynamic, does this check still makes sense?

Reply:
I think it does, this will change in #141 as well, as we would use more percentages than fixed values. But creating a multisig that cannot be used is a user error.